pages/category/[id].vue

```diff
@@ -1,171 +1,136 @@
-      Promise.resolve(siteStore().config || vidoConfig(useRequestHeaders())))
-    const config: VidoConfig = configRef.value
-
-    const fetchSettings = getAsyncDataOrThrows('fetchSettings', () =>
-      siteStore().settings
-        ? Promise.resolve(siteStore().settings as Settings)
-        : getSettings(config))
-
-    const fetchContents = getAsyncDataOrThrows('fetchContents', () =>
-      siteStore().contents
-        ? Promise.resolve(siteStore().contents as ContentEntry[])
-        : getContents(config))
-
-    const fetchPropertyTranslations: Promise<Ref<PropertyTranslations>>
-      = getAsyncDataOrThrows('fetchPropertyTranslations', () =>
-        siteStore().translations
-          ? Promise.resolve(siteStore().translations as PropertyTranslations)
-          : getPropertyTranslations(config))
-
-    const fetchMenuItems = getAsyncDataOrThrows('fetchMenuItems', () =>
-      menuStore().menuItems !== undefined
-        ? Promise.resolve(Object.values(menuStore().menuItems!))
-        : getMenu(config))
-
-    const fetchPoiByCategoryId = getAsyncDataOrThrows(
-      `fetchPoiByCategoryId-${params.id}`,
-      () =>
-        getPoiByCategoryId(config, params.id as string, {
-          geometry_as: 'point',
-          short_description: true,
-        }),
+import { menuStore as useMenuStore } from '~/stores/menu'
+import { siteStore as useSiteStore } from '~/stores/site'
+import Header from '~/components/Layout/Header.vue'
+import Footer from '~/components/Layout/Footer.vue'
+import PoisTable from '~/components/PoisList/PoisTable.vue'
+import CategorySelector from '~/components/PoisList/CategorySelector.vue'
+
+// Query param validation
+definePageMeta({
+  validate({ params }) {
+    return (
+      typeof params.id === 'string' && /^[,-_:a-zA-Z0-9]+$/.test(params.id)
     )
-    const [settings, contents, propertyTranslations, menuItems, pois]
-      = await Promise.all([
-        fetchSettings,
-        fetchContents,
-        fetchPropertyTranslations,
-        fetchMenuItems,
-        fetchPoiByCategoryId,
-      ])
-
-    return {
-      config,
-      settings,
-      contents,
-      propertyTranslations,
-      id: params.id as string,
-      menuItems,
-      pois,
-    }
   },
+})
 
-  computed: {
-    ...mapState(menuStore, ['getCurrentCategory']),
-    ...mapWritableState(siteStore, {
-      locale: 'locale',
-      globalConfig: 'config',
-      globalSettings: 'settings',
-      globalContents: 'contents',
-      globalTranslations: 'translations',
-    }),
-  },
+const router = useRouter()
+const param = useRoute().params
+const id = Number.parseInt(param.id as string)
+const siteStore = useSiteStore()
+const menuStore = useMenuStore()
+const { $vidoConfigSet, $settings, $propertyTranslations, $trackingInit } = useNuxtApp()
 
-  created() {
-    this.globalConfig = this.config
-    if (this.menuItems)
-      menuStore().fetchConfig(this.menuItems)
+// TODO: Get this globally as share it across components / pages
```

Comment:
TODO

Reply:
What do you mean ? You want to do this work within this PR ?
We said later as it will have a global impact. For the moment we scoped changes to the category page.

Reply:
ok